### PR TITLE
dont fail while over writing existing invalid ufo

### PR DESCRIPTION
**Pull request by typemytype**
fe when required files are missing

### CODE REVIEW

**Comment by anthrotype** [comment]
I wonder if maybe there should be a warning when this happens.

**Comment by typemytype** [comment]
whats the name for such a warning?

**Comment by anthrotype** [comment]
I see that we don't use the `logging` module here in defcon. Perhaps then use the `warnings` module, with a `UserWarning` class?

File: Lib/defcon/objects/layerSet.py [hunk 1]
```diff
@@ -283,6 +283,8 @@ def save(self, writer, saveAs=False, progressBar=None):
                     newName = actionData["newName"]
                     if oldName in writer.layerContents:
                         writer.renameGlyphSet(oldName, newName)
+                        if newName == self.defaultLayer.name:
+                            writer.renameGlyphSet(newName, newName, defaultLayer=True)
```
Comment (anthrotype) [inline Lib/defcon/objects/layerSet.py]: why do you have to call `renameGlyphSet` twice in the case of `defaultLayer`, with old and new names both being `newName`?

Comment (anthrotype) [reply Lib/defcon/objects/layerSet.py]: I mean, wouldn't it also work if you did this?

```python
defaultLayer = newName == self.defaultLayer.name
writer.renameGlyphSet(oldName, newName, defaultLayer=defaultLayer)
```

(also this probably belongs to a separate PR, but it's ok)

**Comment by benkiel** [comment]
@anthrotype should we merge this?

**Comment by anthrotype** [comment]
forgot what the issue was.
is this comment relevant? https://github.com/typesupply/defcon/pull/165#discussion_r168920463

feel free to merge if you think it's not

**Comment by benkiel** [comment]
@anthrotype not sure, @typemytype?

**Comment by typemytype** [comment]
the problem is/was that when overwriting an invalid UFO it should not fail to write valid data... instead remove the invalid UFO and write the valid data to disk as a save-as operation.

**Comment by anthrotype** [comment]
fixed by #180

**Pull request closed without merging**